### PR TITLE
[config] Add commands for adding/removing syslog servers

### PR DESCRIPTION
**Pull request by tsvanduyn**
**- What I did**
I've added config commands to add/del syslog servers from the device.

**- How I did it**
```
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ redis-cli -n 4 keys "*" | grep syslog  -i
SYSLOG_SERVER|100.127.20.21
SYSLOG_SERVER|10.3.145.8
SYSLOG_SERVER|4.4.4.4
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ sudo config syslog add 5.5.5.5
Syslog server 5.5.5.5 added to configuration
Restarting rsyslog-config service...
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ redis-cli -n 4 keys "*" | grep syslog  -i
SYSLOG_SERVER|100.127.20.21
SYSLOG_SERVER|10.3.145.8
SYSLOG_SERVER|4.4.4.4
SYSLOG_SERVER|5.5.5.5
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ sudo config syslog del 5.5.5.5
Syslog server 5.5.5.5 removed from configuration
Restarting rsyslog-config service...
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ redis-cli -n 4 keys "*" | grep syslog  -i
SYSLOG_SERVER|100.127.20.21
SYSLOG_SERVER|10.3.145.8
SYSLOG_SERVER|4.4.4.4
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ 

```
**- How to verify it**
```
There are two ways: 
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ redis-cli -n 4 keys "*" | grep syslog  -i
SYSLOG_SERVER|100.127.20.21
SYSLOG_SERVER|10.3.145.8
SYSLOG_SERVER|4.4.4.4

Or 
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ show run all | grep -i syslog -A4
    "SYSLOG_SERVER": {
        "4.4.4.4": {}, 
        "10.3.145.8": {}, 
        "100.127.20.21": {}
    }, 
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ sudo config syslog add 5.5.5.5
Syslog server 5.5.5.5 added to configuration
Restarting rsyslog-config service...
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ show run all | grep -i syslog -A4
    "SYSLOG_SERVER": {
        "4.4.4.4": {}, 
        "5.5.5.5": {}, 
        "10.3.145.8": {}, 
        "100.127.20.21": {}
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ 
```
**- Previous command output (if the output of a command-line utility has changed)**
There was no previous command to add or remove syslog servers. 
**- New command output (if the output of a command-line utility has changed)**
```
config add syslog command:
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ sudo config syslog add 5.5.5.5
Syslog server 5.5.5.5 added to configuration
Restarting rsyslog-config service...
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$

config del syslog command: 
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ sudo config syslog del 5.5.5.5
Syslog server 5.5.5.5 removed from configuration
Restarting rsyslog-config service...
admin@str-s6000-acs-11:/usr/lib/python2.7/dist-packages/config$ 
```
-->

### CODE REVIEW

File: config/main.py [hunk 1]
```diff
@@ -1312,6 +1312,49 @@ def naming_mode_alias():
...
+
+@syslog.command('add')
+@click.argument('syslog_ip_address', metavar='<syslog_ip_address>', required=True)
+@click.pass_context
+def add_syslog_server(ctx, syslog_ip_address):
+    """ Add syslog server IP """
+    if not is_ipaddress(syslog_ip_address):
+        ctx.fail('Invalid ip address')
+    db = ctx.obj['db']
+    db.set_entry('SYSLOG_SERVER', syslog_ip_address, {'NULL': 'NULL'})
```
Comment (jleveque) [inline config/main.py]: You should first check to see if the IP we're adding already exists.

Comment (tsvanduyn) [reply config/main.py]: Added the check to confirm if IP already exists.

File: doc/Command-Reference.md [hunk 2]
```diff
@@ -4079,4 +4079,35 @@ This command displays the routing policy that takes precedence over the other ro
 		Exit routemap
   ```
 
+**Syslog config commands 
+
+This sub-section of commands is used to add or remove the configured syslog servers.
+
+**config syslog add
```
Comment (jleveque) [inline doc/Command-Reference.md]: Surround entire command with `**` to make it bold to match style of rest of document (`**config syslog add**`).

File: doc/Command-Reference.md [hunk 3]
```diff
@@ -4079,4 +4079,35 @@ This command displays the routing policy that takes precedence over the other ro
...
+- Usage: config syslog add <ip-address>
+- Example: 
+  ```
+  admin@str-s6000-acs-11:~$ sudo config syslog add 1.1.1.1
+  Syslog server 1.1.1.1 added to configuration
+  Restarting rsyslog-config service...
+  admin@str-s6000-acs-11:~$
+  ```
+
+**config syslog delete
```
Comment (jleveque) [inline doc/Command-Reference.md]: Surround entire command with `**` to make it bold to match style of rest of document (`**config syslog delete**`).

Comment (tsvanduyn) [reply doc/Command-Reference.md]: fixed

File: doc/Command-Reference.md (hunk 2 above)
Comment (tsvanduyn) [reply doc/Command-Reference.md]: fixed.

**Comment by jleveque** [comment]
Retest this please

File: config/main.py [hunk 4]
```diff
@@ -1312,6 +1312,54 @@ def naming_mode_alias():
     """Set CLI interface naming mode to ALIAS (Vendor port alias)"""
     set_interface_naming_mode('alias')
 
+#
+# 'syslog' group ('config syslog ...')
+#
+@config.group()
+@click.pass_context
+def syslog(ctx):
```
Comment (stephenxs) [inline config/main.py]: It seems that the function name "syslog" conflicts with the module "syslog", causing functions calling logxxx failed. 
Is it possible to rename it?

Comment (nazariig) [reply config/main.py]: @stephenxs / @jleveque the ticket: https://github.com/Azure/sonic-utilities/issues/635

**Pull request merged**